### ISSUE TITLE
Social provider - no way of knowing if login or signup

### ISSUE THREAD

**Issue by sagivf** [driftyco/ionic-cloud#22]
When I use a social provider there is no way to know if the user has just signed up for the first time or logged in. There should be some parameter in the callback/promise informing this. 

I could create two separate views but:
a) Bad UX
b) The user might have signed up in the past and there is no easy way of telling.

**Comment by dwieeb** [comment]
@sagivf I also noticed this while playing with our social auth. @devnetics, want to work with me next week to ensure this functionality? Should be pretty simple.

**Comment by dwieeb** [comment]
We'll be including this in the milestone. @devnetics is making some good progress with this.

**Comment by dwieeb** [comment]
@sagivf If you update, the signup flag is now in the object resolved from `login()`.

**Comment by sagivf** [comment]
Thanks @dwieeb looks good, I was a little confused at first as this only applies to social signup/login. but that makes sense - login with an already registered user, does and should be rejected. I would make this clear in the docs and perhaps change the `signup?` param to `socialSignup?` to be clear.

**Comment by dwieeb** [comment]
@sagivf It shouldn't be too confusing, right? Calling `signup()` is only for email/password authentication. For social authentication (and custom authentication, in which case a flag named `socialSignup?` would misrepresent), you only need to call `login('facebook')`. The `'facebook'` would indicate which auth provider is used, wouldn't it? 

I certainly can return the `AuthModuleId` (e.g. `'facebook'`, `'basic'`, etc.) in the new `AuthLoginResult`, though, if it is helpful.

**Comment by sagivf** [comment]
@dwieeb maybe we should wait for more feedback. I just think that users that haven't used social login will get confused by the differences, I suppose just having clear docs will solve it.

Adding  the `AuthModuleId` would be nice - not critical.

**Comment by dwieeb** [comment]
@sagivf Honestly, I'd like to make aggressive changes to the User/Auth cloud client API itself. I'd like to split `Auth` into classes such as `EmailPasswordAuth`, `FacebookAuth`, etc. such that the methods are relevant to each auth type and their inputs and outputs are scoped to their authentication scheme. `FacebookAuth` would set the active user as a `FacebookUser` with attributes relevant to Facebook, instead of digging in `user.social.facebook.profile_picture`, for example.

Maybe I should do this before we launch the new documentation so that it's more clear right from the start. Thoughts?

**Comment by sagivf** [comment]
@dwieeb Heres my thoughts.

Firstly - if this were not in beta I would defiantly say it's not worth a braking change.

As far as the relevant attributes - I would in any case try to consolidate them into common attraibutes on the user (with an option to get unique ones).

If I'm not missing anything It's basically just a syntax change:
`this.auth.login('facebook')`
`this.auth.login('basic', { name: '', password: '' })`
vs
`this.auth. FacebookAuth()`
`this.auth. EmailPasswordAuth({ name: '', password: '' })`

I think the later is easier to understand since it makes it clear the EmailPassword is different then then FacebookAuth. However the original one makes you like less code and helps in reducing the amount of code. The later is also better typed (no string passing) but I still don't think Typing is the most important thing.

Bottom line, I don't think i would change it, but it's not clear cut and both are good.
